roofit/roofitcore/inc/RooTemplateProxy.h

```diff
@@ -176,7 +176,7 @@ class RooTemplateProxy : public RooArgProxy {
   /// \param[in] proxyOwnsArg Proxy will delete the payload if owning.
   RooTemplateProxy(const char* theName, const char* desc, RooAbsArg* owner, T& ref,
       Bool_t valueServer=true, Bool_t shapeServer=false, Bool_t proxyOwnsArg=false) :
-        RooArgProxy(theName, desc, owner, ref, valueServer, shapeServer, proxyOwnsArg) { }
+        RooArgProxy(theName, desc, owner, const_cast<typename std::remove_const<T>::type&>(ref), valueServer, shapeServer, proxyOwnsArg) { }
```

Comment:
Can you please write either in the commit message or here in the PR conversation what the motivation is for this change? Thanks a lot!

Reply:
I changed the proxy such that you can also instantiate it as
```c++
RooTemplateProxy<const Stuff>
```

Now, the base class, `RooArgProxy` is not a template, and it only ever takes `RooAbsArg*` (non-const!). This cannot change (unless you are willing to go through a lot of pain).
The type and const correctness therefore only comes in through the casts that happen in `RooTemplateProxy`, but in order to hand a `const` reference up to `RooArgProxy`, you have to cast the constness away.

See another way to do it?

Reply:
I see, so if one uses `RooTemplateProxy` as intended, the cons-correctnes violation is just behind the scenes. Makes sense!